### PR TITLE
[B2BP-401] Add OneTrust section to render Privacy Policy

### PR DESCRIPTION
**Pull request by simosalsi**
Depends on #215 

As per title.

A section has been added so that we're able to render a [privacy policy page has seen in SEND's website](https://notifichedigitali.pagopa.it/informativa-privacy/).

#### List of Changes

- Added `OneTrust` section to Strapi --> It currently contains only the single "cdnURL" field and is treated has any other section. The approach may change upon feedback from end users.
- Added OneTrust's script to `/public` folder
- Load the above mentioned script in `Layout.tsx` to avoid duplication and conflicts
- Added `OneTrustSection` component to NextJS --> This component loads the specific cdnURL and has intentionally been kept client-side to make sure the page fetches the latest updated privacy policy every time, instead of at build time.

#### Motivation and Context

Replicate SEND's website in its entirety.

#### How Has This Been Tested?

Tested locally.

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/nextjs-website/src/components/OneTrust.tsx
Comment (datalek) [inline apps/nextjs-website/src/components/OneTrust.tsx:1]: @simosalsi What do you think about add a test to this component?

File: apps/nextjs-website/src/components/OneTrust.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  // eslint-disable-next-line functional/no-return-void
+  useEffect(() => {
+    // eslint-disable-next-line functional/no-return-void
+    OneTrust.NoticeApi.Initialized.then(() => {
+      OneTrust.NoticeApi.LoadNotices([cdnURL], false);
+    });
+  }, [cdnURL]);
+
+  return <div id={divID} />;
```
Comment (datalek) [inline apps/nextjs-website/src/components/OneTrust.tsx:36]: @simosalsi When is the `id` used? Can we omit it?

File: apps/strapi-cms/src/components/sections/one-trust.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "collectionName": "components_sections_one_trusts",
+  "info": {
+    "displayName": "OneTrust"
+  },
+  "options": {},
+  "attributes": {
+    "cdnURL": {
+      "type": "string",
+      "required": true
```
Comment (datalek) [inline apps/strapi-cms/src/components/sections/one-trust.json:10]: @simosalsi What does this `cdnURL` should contain? The name confuses me a bit at the moment

File: apps/nextjs-website/src/components/OneTrust.tsx
Comment (simosalsi) [reply apps/nextjs-website/src/components/OneTrust.tsx:1]: What would you test exactly?

Only thing worth testing to me would be the IDFromURL function.

File: apps/nextjs-website/src/components/OneTrust.tsx (hunk 1 above)
Comment (simosalsi) [reply apps/nextjs-website/src/components/OneTrust.tsx:36]: OneTrust's script looks for that specific id when it's trying to determine which div it has to populate, so it is mandatory.

File: apps/nextjs-website/src/components/OneTrust.tsx
Comment (datalek) [reply apps/nextjs-website/src/components/OneTrust.tsx:1]: Exactly, the `IDFromURL` function at least

File: apps/nextjs-website/src/components/OneTrust.tsx (hunk 1 above)
Comment (datalek) [reply apps/nextjs-website/src/components/OneTrust.tsx:36]: And how the OneTrust script know the id to look for?

File: apps/strapi-cms/src/components/sections/one-trust.json (hunk 2 above)
Comment (datalek) [reply apps/strapi-cms/src/components/sections/one-trust.json:10]: What about renaming it to `oneTrustURL` or something like that?

File: apps/nextjs-website/src/components/OneTrust.tsx (hunk 1 above)
Comment (simosalsi) [reply apps/nextjs-website/src/components/OneTrust.tsx:36]: It extrapolates it from `cdnURL` akin to how we do it inside the `IDFromURL` function

File: apps/nextjs-website/src/components/OneTrust.tsx
Comment (simosalsi) [reply apps/nextjs-website/src/components/OneTrust.tsx:1]: Will add the test ASAP

File: apps/strapi-cms/src/components/sections/one-trust.json (hunk 2 above)
Comment (simosalsi) [reply apps/strapi-cms/src/components/sections/one-trust.json:10]: It's already inside the `OneTrust` component so it seems redundant imo

File: apps/nextjs-website/src/components/OneTrust.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+const OneTrustSection = (props: OneTrustSectionProps) => {
+  const { cdnURL } = props;
+  const divID = `otnotice-${IDFromURL(cdnURL)}`;
+
+  // eslint-disable-next-line functional/no-return-void
+  useEffect(() => {
+    // eslint-disable-next-line functional/no-return-void
+    OneTrust.NoticeApi.Initialized.then(() => {
+      OneTrust.NoticeApi.LoadNotices([cdnURL], false);
```
Comment (datalek) [inline apps/nextjs-website/src/components/OneTrust.tsx:32]: What about adding a comment that explain what this code does?
```suggestion
    OneTrust.NoticeApi.Initialized.then(() => {
      // The below command looks for a div with id ... 
      OneTrust.NoticeApi.LoadNotices([cdnURL], false);
```

File: apps/nextjs-website/src/components/OneTrust.tsx (hunk 1 above)
Comment (datalek) [inline apps/nextjs-website/src/components/OneTrust.tsx:36]: Same here, what about adding a comment that explain what this code does?
```suggestion
  // This div is going to be populated by ...
  return <div id={divID} />;
```

File: apps/strapi-cms/src/components/sections/one-trust.json (hunk 2 above)
Comment (simosalsi) [reply apps/strapi-cms/src/components/sections/one-trust.json:10]: After a quick call, will update to `oneTrustNoticeURL`

File: apps/nextjs-website/src/lib/oneTrust.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+export const ExtractNoticeIDFromOneTrustURL = (
```
Comment (datalek) [inline apps/nextjs-website/src/lib/oneTrust.ts:1]: **suggestion**: What about lowercase it? If you do so you should change the references
```suggestion
export const extractNoticeIDFromOneTrustURL = (
```

File: apps/nextjs-website/src/components/OneTrust.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+declare const OneTrust: {
+  NoticeApi: {
+    Initialized: {
+      // eslint-disable-next-line functional/no-return-void
+      then: (cbk: () => void) => void;
+    };
+    // eslint-disable-next-line functional/no-return-void
+    LoadNotices: (noticesUrls: string[], flag: boolean) => void;
+  };
+};
```
Comment (datalek) [inline apps/nextjs-website/src/components/OneTrust.tsx:16]: What about moving this declaration into `src/lib/oneTrust.ts`?

**Review by datalek — APPROVED** [review]
Just a couple of non-blocking suggestions, feel free to incorporate them or not

**Pull request merged**